test/unit/cmap/auth/mongodb_oidc/token_entry_cache.test.ts

```diff
@@ -0,0 +1,143 @@
+import { expect } from 'chai';
+
+import { TokenEntryCache } from '../../../../mongodb';
+
+describe('TokenEntryCache', function () {
+  const tokenResult = {
+    accessToken: 'test',
+    expiresInSeconds: 100
+  };
+
+  const serverResult = {
+    clientId: 1
+  };
+
+  describe('#addEntry', function () {
+    context('when expiresInSeconds is provided', function () {
+      const cache = new TokenEntryCache();
+      let entry;
+
+      before(function () {
+        cache.addEntry('localhost', 'user', tokenResult, serverResult);
+        entry = cache.getEntry('localhost', 'user');
+      });
+
+      it('sets the username in the key', function () {
+        expect(cache.entries.has('localhost-user')).to.be.true;
+      });
+
+      it('adds the token result', function () {
+        expect(entry.tokenResult).to.deep.equal(tokenResult);
+      });
+
+      it('adds the server result', function () {
+        expect(entry.serverResult).to.deep.equal(serverResult);
+      });
+
+      it('creates an expiration', function () {
+        expect(entry.expiration).to.be.above(Date.now());
```

Comment:
I think we could narrow the assertion range here using the `within` operator, or with a companion `below`

Reply:
Changed assertion to be `at.most` for the `Date.now + expiresInSeconds * 1000`

Reply:
there is a typo in the new line, but I added a new comment, so resolving this one